types/collections.go

```diff
@@ -135,6 +135,28 @@ func LengthPrefixedAddressKey[T addressUnion](keyCodec collcodec.KeyCodec[T]) co
 	}
 }
 
+type legacyKeyCodec[T []byte] struct {
+	collcodec.KeyCodec[T]
```

Comment:
Since the type parameter `T`  is always `[]byte` we can remove the type parameter. And simply use `[]byte`  everywhere.